<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Utiliser les exceptions</title>
<link rel="stylesheet" type="text/css" href="dbstyle.css">
<meta name="generator" content="DocBook XSL Stylesheets V1.79.2">
<link rel="home" href="index.html" title="Guide de référence du programmeur">
<link rel="up" href="zend.exception.html" title="Zend_Exception">
<link rel="prev" href="zend.exception.html" title="Zend_Exception">
<link rel="next" href="zend.exception.basic.html" title="Utilisation classique">
</head>
<body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF">
<div class="navheader"><table width="100%" summary="Navigation header">
<tr><th colspan="3" align="center">Utiliser les exceptions</th></tr>
<tr>
<td width="20%" align="left">
<a accesskey="p" href="zend.exception.html">Précédent</a> </td>
<th width="60%" align="center">Zend_Exception</th>
<td width="20%" align="right"> <a accesskey="n" href="zend.exception.basic.html">Suivant</a>
</td>
</tr>
</table></div>
<div class="sect1">
<div class="titlepage"><div><div><h2 class="title" style="clear: both">
<a name="zend.exception.using"></a>Utiliser les exceptions</h2></div></div></div>
    

    <p>
        <code class="classname">Zend_Exception</code> est la classe de base dont dérivent toutes les
        exceptions levées par les classes de Zend Framework.
    </p>

    <div class="example">
<a name="zend.exception.using.example"></a><p class="title"><b>Exemple 385. Récupération d'une exception</b></p>
<div class="example-contents">
        

        <p>
            Le code suivant montre comment attraper une exception levée par une classe de
            Zend Framework :
        </p>

        <pre class="programlisting">
try {
    Zend_Loader::loadClass('classnonexistante');
} catch (Zend_Exception $e) {
    // Appeler Zend_Loader::loadClass() sur une classe non-existante
    //entrainera la levée d'une exception dans Zend_Loader
    echo "Récupère exception: " . get_class($e) . "\n";
    echo "Message: " . $e-&gt;getMessage() . "\n";
    // puis tout le code nécessaire pour récupérer l'erreur
}
</pre>

    </div>
</div>
<br class="example-break">

    <p>
        <code class="classname">Zend_Exception</code> peut être comme une classe d'exception
        catch-all dans un bloc catch pour traquer toues les exceptions levées par les classes
        de Zend Framework. Ceci peut être utile quand un programme n'arrive pas à fonctionner en
        essayant d'attraper un type d'exception spécifique.
    </p>

    <p>
        La documentation pour chaque composant de Zend Framework et de ses classes contient
        les informations plus spécifiques sur les méthodes qui lèvent des exceptions, les
        circonstances de lancement de ces exceptions et quelles types declasses d'exception peuvent
        être levées.
    </p>

</div>
<div class="navfooter"><table width="100%" summary="Navigation footer">
<tr>
<td width="40%" align="left">
<a accesskey="p" href="zend.exception.html">Précédent</a> </td>
<td width="20%" align="center"><a accesskey="u" href="zend.exception.html">Niveau supérieur</a></td>
<td width="40%" align="right"> <a accesskey="n" href="zend.exception.basic.html">Suivant</a>
</td>
</tr>
<tr>
<td width="40%" align="left" valign="top">Zend_Exception </td>
<td width="20%" align="center"><a accesskey="h" href="index.html">Sommaire</a></td>
<td width="40%" align="right" valign="top"> Utilisation classique</td>
</tr>
</table></div>
<div class="revinfo"></div>
</body>
</html>
